### PR TITLE
Enable support for Python3.11

### PR DESCRIPTION
**Pull request by pdthummar**
### Issue describing the changes in this PR

Adding support for python3.11

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs [hunk 1]
```diff
@@ -165,31 +165,31 @@ public static void AssertPythonVersion(WorkerLanguageVersionInfo pythonVersion,
         {
             if (pythonVersion?.Version == null)
             {
-                var message = "Could not find a Python version. Python 3.6.x, 3.7.x, 3.8.x, 3.9.x, or 3.10.x is recommended, and used in Azure Functions.";
+                var message = "Could not find a Python version. Python 3.6.x, 3.7.x, 3.8.x, 3.9.x, 3.10.x or 3.11.x is recommended, and used in Azure Functions.";
```
Comment (vrdmr) [inline src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:168]: Lets remove Python 3.6 from the list. Its a v3 specific thing

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs [hunk 2]
```diff
@@ -165,31 +165,31 @@ public static void AssertPythonVersion(WorkerLanguageVersionInfo pythonVersion,
...
                 if (errorIfNoVersion) throw new CliException(message);
                 ColoredConsole.WriteLine(WarningColor(message));
                 return;
             }
 
             ColoredConsole.WriteLine(AdditionalInfoColor($"Found Python version {pythonVersion.Version} ({pythonVersion.ExecutablePath})."));
 
-            // Python 3.[6|7|8|9|10] (supported)
+            // Python 3.[6|7|8|9|10|11] (supported)
```
Comment (vrdmr) [inline src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:176]: 3.6 references everywhere

Comment (vrdmr) [reply src/Azure.Functions.Cli/Helpers/PythonHelpers.cs:176]: Unless there's something I miss

File: src/Azure.Functions.Cli/StaticResources/StaticResources.cs [hunk 3]
```diff
@@ -49,6 +49,8 @@ public static async Task<string> GetValue(string name)
         
         public static Task<string> DockerfilePython310 => GetValue("Dockerfile.python3.10");
 
+        public static Task<string> DockerfilePython311 => GetValue("Dockerfile.python3.10");
+
```
Comment (vrdmr) [inline src/Azure.Functions.Cli/StaticResources/StaticResources.cs:53]: 3.10 -> 3.11

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs (hunk 1 above)
Comment (pdthummar) [reply src/Azure.Functions.Cli/Helpers/PythonHelpers.cs]: removed. thanks

File: src/Azure.Functions.Cli/Helpers/PythonHelpers.cs (hunk 2 above)
Comment (pdthummar) [reply src/Azure.Functions.Cli/Helpers/PythonHelpers.cs]: removed. thanks

File: src/Azure.Functions.Cli/StaticResources/StaticResources.cs (hunk 3 above)
Comment (pdthummar) [reply src/Azure.Functions.Cli/StaticResources/StaticResources.cs:53]: Updated. thanks

**Comment by pdthummar** [comment]
Tested generated build by running python 3.11 V1 and V2 functions locally.

**Review by vrdmr — APPROVED** [review]

**Review by khkh-ms — APPROVED** [review]
LGTM

**Pull request merged**